File: src/eQuantic.UI.Runtime/src/shared/__transpiled__/Tooltip.ts
import { Anchored, AnchorPlacementValue, Box, BoxStyle, BuildContext, CornerRadii, EdgeInsets, StatelessComponent, Text, VisualNode } from "@equantic/runtime";

export class Tooltip extends StatelessComponent {
    declare child: VisualNode;
    declare text: string;
    declare placement: AnchorPlacementValue;
    constructor(child?: any, text?: any, props?: any) {
        super();
        if (child !== undefined) this.child = child;
        if (text !== undefined) this.text = text;
        if (this.placement === undefined) this.placement = 'topCenter';
        this.child = child;
        this.text = text;
        if (props && typeof props === 'object') Object.assign(this, props);
    }

    build(context: BuildContext) {
        let theme = context.theme;
        let pill = new Box(new BoxStyle({ background: theme.textPrimary, cornerRadius: new CornerRadii(theme.shape('small')), padding: EdgeInsets.symmetric(8, 4) }), new Text(this.text, 'caption', theme.textInverse, 1));
        return new Anchored(this.child, pill, { placement: this.placement, openOnHover: true, describesAnchor: true });
    }

}

File: src/eQuantic.UI.Components/Tooltip.cs
using eQuantic.UI.Primitives;

namespace eQuantic.UI.Components;

/// <summary>
/// The design system's Tooltip (wave 3b): a short hint on an INVERSE pill (TextPrimary fill /
/// TextInverse caption — the Toast pair, flipping correctly in both modes) that shows while the
/// pointer hovers the child. Pure mechanics on both targets: the Anchored hover reveal is CSS-only
/// on web (zero JS, never on touch) and rides the host's hover pipeline on Photon. Keep it to a few
/// words — tooltips are hints, not documentation. Keyboard FOCUS reveals it exactly like hover
/// (:has(:focus-visible), still zero JS), and the panel is a real <c>role="tooltip"</c> the anchor
/// points at with <c>aria-describedby</c> — a screen reader hears the hint after the control's
/// name whether or not anything is visually revealed.
/// v1 fences: show/hide delay, arrow caret, Esc suppression (WCAG 1.4.13's dismissable clause —
/// joins the Anchored Esc work, one controller for every floating panel).
/// </summary>
public sealed class Tooltip : StatelessComponent
{
    public Tooltip(VisualNode child, string text)
    {
        Child = child;
        Text = text;
    }

    public VisualNode Child { get; init; }
    public string Text { get; init; }
    public AnchorPlacement Placement { get; init; } = AnchorPlacement.TopCenter;

    public override VisualNode Build(ComponentContext context)
    {
        var theme = context.Theme;

        var pill = new Box(new BoxStyle
        {
            Background = theme.TextPrimary,
            CornerRadius = new CornerRadii(theme.Shape(ShapeScale.Small)),
            Padding = EdgeInsets.Symmetric(Space.S2, Space.S1),
        }, new Text(Text, TypeRole.Caption, theme.TextInverse, maxLines: 1));

        return new Anchored(Child, pill)
        {
            Placement = Placement,
            OpenOnHover = true,
            DescribesAnchor = true,
        };
    }
}
